### DISCUSSION TITLE
Option to Export to CSV

### DISCUSSION THREAD

**Discussion by amitpujar** [community/community#6584] [Projects and Issues] [answered]
Hello Github,

**The TABLE view is wonderful. Thank you!**

It will be great if you can add an option to export this table view to CSV. 

Use case: As with most management processes, program managers and project managers will need the list copied and pasted in other tools such as PPTs for presentations to senior management.

Thank you,
Amit

**Comment by azenMatt** [comment]
Hi @amitpujar - we have a work around for this, if you try: 

- Hit cmd+a twice (this will select all)
- Hit cmd+c (copy) 
- Head to your spreadsheet of choice
- Hit cmd+v (paste) 

You'l then have the contents of the table in CSV format. Which you can then play with the format options to turn it into a spreadsheet if you want

**Reply by amitpujar** [reply]
Thank you, that works. But needs a manual cleanup.

I use cli to get the CSV and then in Excel, use the 'Text to Columns' option to achieve this.

**Reply by lrjbrual** [reply]
This seems work. but I think it is also better if you can just download the CSV so their is no additional work around todo.

**Reply by nasrulhazim** [reply]
> Thank you, that works. But needs a manual cleanup.
> 
> I use cli to get the CSV and then in Excel, use the 'Text to Columns' option to achieve this.

yerp. still need clean up, need to transpose the result so that it can properly display in rows n columns.

**Reply by lkerxhalli** [reply]
If you are running into a problem where cmd+c is not working, make sure that the browser copy is not overriding the javascript shortcut for copying the selected rows. If you see the Edit menu lighting up when you use the cmd+c, it means that the javascript is not copying it. This started to work for me after few tries.

**Reply by kathy-phet** [reply]
@azenMatt - I can see the rows highlight, but cntl-c doesn't seem to be copying anything to the clip board. So this workaround isn't working. An true export to CSV function would be a great improvement.

**Reply by dmarcey** [reply]
@kathy-phet - we identified the bug with copy / paste and rolled out a fix late last week. Please let us know if the work-around identified above is still not working for you. Sorry for the disruption!

**Reply by josteinaj** [reply]
In Ubuntu, copying works from Chromium but not Firefox. The result is unstructured text (no straight-forward way to recreate the table by pasting in LibreOffice or splitting on every Nth line in a text editor). If I try inspecting the DOM and copy the HTML from the table, I only get a little over 20 rows at a time, depending on what is visible on the screen. A proper CSV export would be very welcome!

**Reply by josteinaj** [reply]
There seems to be a limit to the number of rows when using select all + copy as well.  I get around 40 lines using that method.

If I only need one column exported, I've found the most efficient way to get it exported is to (1) use Chromium, (2) hide all other columns than the one needed, (3) use the select all + copy method to export rows, and (4) repeat for each set of 20-40 rows (i.e. what is visible on screen). (5) Finally do some manual cleanup to avoid duplicate rows etc.

**Reply by josteinaj** [reply]
Actually, if I just zoom out far enough so that all rows are visible on screen at once, I can copy the HTML from the DOM and extract data from each div with class=row, and thus are able to copy all the rows in one go, and in a structured format. Zooming out is probably a good idea for the select all + copy workaround as well.

**Reply by azenMatt** [reply]
👍 

I've unmarked this an answer as even though it is a work-around... it's not really what is needed. Agreed we do need a full export to csv solution here. Having it as unanswered will help me see how it is trending against other open requests.

**Reply by LucyRussell99** [reply]
It doesn't work for me, I don't know why.  Currently, not having an export to CSV that works is a big blocker for our company moving over to using this feature, I do all my case reporting in Excel/external systems and I need some sort of CSV output to continue this, it would make a huge difference if it was available.

**Comment by rurara** [comment]
![image](https://user-images.githubusercontent.com/2038631/142474532-209e4e09-ac81-4a49-9b15-66d088cccd54.png)

I want this feature too
If the user name does not appear and the issue contains "," the field changes

**Comment by kevintich** [comment]
I support this. Would be a good to have

**Comment by danm** [comment]
hey @azenMatt, this is now closed due to the copy and paste workaround - but I would love something better than that, especially as our delivery person always wants to export everything into one place.

Does the fact the discussion is closed mean it wouldn't be considered for further improvement?

**Reply by azenMatt** [reply]
>Does the fact the discussion is closed mean it wouldn't be considered for further improvement?

No, I would def consider having first class export to CSV as something we would like to have in the future. I just wanted to mark the workaround above as the 'answer' to make it easier to find for anyone else looking for a solution

**Comment by jfritz** [comment]
I echo the need for a CSV export. CSV export is a first-class feature in other Agile tools that have robust backlog management. Ideally it would export the exact view filtered as-is and not a full raw data dump. This is useful for those of us that want to do advanced custom reporting using exports to Excel/LibreOffice Calc, etc.

**Reply by jfritz** [reply]
It's incredible to me how long this has gone without being implemented as a first class feature. It really speaks to a lack of consideration for product owner-centric and scrummaster-centric features.

**Comment by orothe** [comment]
cmd+a twice is a good workaround.
But the "assignees" got lost.

**Comment by lesleyauk** [comment]
I'm not sure since when but we noticed last week that the workaround from @azenMatt no longer works: "Copy" adds _nothing_ to the clipboard.

While we are waiting for other features to be added, using a spreadsheet for additional manipulation and analysis was a key part of our workflow. Hope this can be fixed again soon!

**Reply by azenMatt** [reply]
seems like a 🐛 - we are taking a look

**Reply by dmarcey** [reply]
Thanks for the bug report @lesleyauk! We landed a fix for this bug late last week. Sorry about the disruption and please let us know if you continue to experience this issue!

**Comment by kathy-phet** [comment]
@azenMatt - Same issue as @lesleyauk above. The solution is not working for me either as noted 3 weeks ago. 

It seems like the "Marked as answered" label needs to be removed, if the temporary solution isn't work? 

I would love this feature! Bringing the data in the sheets into something like Excel is an important part of our workflow too, and seems like key base functionality for these new project tables.

**Comment by LucyRussell99** [comment]
Export is needed please!  This workaround is not great as the format it results in is not useful - a lot of work to get it usable.

**Comment by ianfortier** [comment]
The CTRL+C / CTRL+V doesn't work on Windows :/

**Comment by kathy-phet** [comment]
Thank you for unmarking it as answered! It is a key feature for us.

**Comment by joachimweyl** [comment]
Please include this feature.
+1

**Comment by agarciacalvo** [comment]
+1 :)

**Comment by bala-techguy** [comment]
We recently started using this feature from Github and it will be great if a bare minimum export option is given to start with. It need not be a fancy export option to start with.

**Comment by malkoG** [comment]
+1 :)

**Comment by rileybroughten** [comment, marked as answer]
You can now export your project view to a CSV file! Check it out on the [changelog](https://github.blog/changelog/2023-06-15-github-issues-projects-june-15th-update/#%F0%9F%8F%97%EF%B8%8F-export-project-view-as-a-csv-file) announcement and let us know what you think 🚀

**Reply by amitpujar** [reply]
This is **wonderful news**. Thank you, **Riley and the team** for implementing this.
